### ISSUE TITLE
Support MS SQL Server

### ISSUE THREAD

**Issue by marpio** [gobuffalo/pop#40]
It's an common DB in the enterprise and now also available for Linux.

**Comment by markbates** [comment]
Completely agree, it just needs someone to step up and implement it. :)

**Comment by marpio** [comment]
Well, I'm interested. Although I probably would need some guidance.

**Comment by markbates** [comment]
Of course! I’m happy to offer guidance. @mclark4386 implemented cockroach support recently. I’m sure he’ll be happy to offer some guidance along the way too!

**Comment by marpio** [comment]
@markbates  So I took a glance at the cockroach commits and if I understand correctly the main task would be to implement sqlserver.go and fizz/translators/sqlserver.go similar to mysql/postgres/cockroach. Is that right?

**Comment by markbates** [comment]
That’s the basics of it, yep. I would pick the implementation that’s closest to MS, I’ve never used it, and use that as a starting point. Rename a few things, point them at the MS DB and see what the tests yell at you about.

**Comment by marpio** [comment]
@markbates  hmm... travis doesn't support ms sql server. would you consider to add appveyor integration?
https://www.appveyor.com/

**Comment by markbates** [comment]
Sure. We use it for Buffalo too. Travis also supports docker, if that helps.

-----------
Mark Bates

**Comment by marpio** [comment]
@markbates true - i've found this: https://github.com/aspnet/EntityFrameworkCore/pull/7842/files
I'll go with travis then.

**Comment by mclark4386** [comment]
@marpio happy to help if you have questions! Sounds like you have a good start going.

**Comment by marpio** [comment]
@markbates short question - what is the purpose of the {db}_meta.go files in the fiz/translators package? Why doesn't postgres need it?

**Comment by markbates** [comment]
Some of the translators, like SQLite (for example), need additional meta information about the table(s), indexes, etc… to create the proper sql for things such as renaming a column. 

Postgres is a nice db and things just work. :)

**Comment by marpio** [comment]
@markbates thanks

**Comment by marpio** [comment]
I'm having some difficulties - sqlx doesn't support the sqlserver parameter syntax so:
```
// this doesn't work
tx.MustExec("INSERT INTO person (first_name, last_name, email) VALUES ($1, $2, $3)", "Jason", "Moiron", "jmoiron@jmoiron.net")
// this works
tx.MustExec("INSERT INTO person (first_name, last_name, email) VALUES (@p1, @p2, @p3)", "Jason", "Moiron", "jmoiron@jmoiron.net")

// but this doesn't
tx.NamedExec("INSERT INTO person (first_name, last_name, email) VALUES (:first_name, :last_name, :email)", &Person{"Jane", "Citizen", "jane.citzen@example.com"}) 
// neither does this
tx.NamedExec("INSERT INTO person (first_name, last_name, email) VALUES (@first_name, @last_name, @email)", &Person{"Jane", "Citizen", "jane.citzen@example.com"}) 
```
there is this issue jmoiron/sqlx#374 regarding this problem but it doesn't seem to be active.

The second problem is that sqlserver doesn't support the usual LIMIT and OFFSET syntax so i would need to somehow override the `buildPaginationClauses` function...

Any ideas how this could be solved?

**Comment by marpio** [comment]
Waiting for jmoiron/sqlx#406

**Comment by stanislas-m** [comment]
@marpio jmoiron/sqlx#406 was merged, and I just merged your PR in fizz. :)

**Comment by marpio** [comment]
Awesome @stanislas-m I will need some time to make it work since a lot changed in buffalo/pop since March.

**Comment by stanislas-m** [comment]
No problem, take your time!

**Comment by Norris1z** [comment]
@marpio any update on this?

**Comment by marpio** [comment]
@Norris1z
I'm quite busy these days and not sure when I get to it. Probably in 2-3 months. Sorry for that. Feel free to take this one if you're interested!

**Comment by andyedison** [comment]
I'm interesting in this. I'm a little green when it comes to Go, and would want to chat about what the scope of work looks like before volunteering to take this on. Gopher slack the best place for that?

**Comment by stanislas-m** [comment]
@andyedison Sure, feel free to join the `buffalo-dev` channel on Gopher slack. :)
